tests/drivers/stepper/stepper_api/src/main.c

```diff
@@ -80,8 +83,13 @@ ZTEST_F(stepper, test_target_position)
 	int32_t pos = 100u;
 
 	(void)stepper_set_max_velocity(fixture->dev, 100u);
-	(void)stepper_set_callback(fixture->dev, fixture->callback, NULL);
+
+	/* Pass the function name as user data */
+	void *user_data_set = (void *)__func__;
+	(void)stepper_set_callback(fixture->dev, fixture->callback, user_data_set);
```

Comment:
user data should be static (which this might technically be since its pointing to a char str, but its quite obfuscated), or the `stepper_set_callback` should be used to clear the callback after (or as part of) the test (set callback to NULL).

Simplest solution is to make the fixture "global" and just set that as the user data:
zassert_equal(user_data, &fixture, "User data not correct");